src/test/java/org/elasticsearch/bwcompat/OldIndexBackwardsCompatibilityTests.java

```diff
@@ -258,9 +273,52 @@ public void testOldIndexes() throws Exception {
         }
     }
 
+    @Test
+    public void testHandlingOfUnsupportedDanglingIndexes() throws Exception {
+        setupCluster();
+        Collections.shuffle(unsupportedIndexes, getRandom());
+        for (String index : unsupportedIndexes) {
+            assertUnsupportedIndexHandling(index);
+        }
+    }
+
+    /**
+     * Waits for the index to show up in the cluster state in closed state
+     */
+    void ensureClosed(final String index) throws InterruptedException {
+        assertTrue(awaitBusy(new Predicate<Object>() {
+            @Override
+            public boolean apply(Object o) {
+                ClusterState state = client().admin().cluster().prepareState().get().getState();
+                return state.metaData().hasIndex(index) && state.metaData().index(index).getState() == IndexMetaData.State.CLOSE;
+            }
+        }));
+    }
+
+    /**
+     * Checks that the given index cannot be opened due to incompatible version
+     */
+    void assertUnsupportedIndexHandling(String index) throws Exception {
+        long startTime = System.currentTimeMillis();
+        logger.info("--> Testing old index " + index);
+        String indexName = loadIndex(index);
+        // force reloading dangling indices with a cluster state republish
+        client().admin().cluster().prepareReroute().get();
+        ensureClosed(indexName);
+        try {
+            client().admin().indices().prepareOpen(indexName).get();
+            fail("Shouldn't be able to open an old index");
+        } catch (IndexException ex) {
+            assertThat(ex.getMessage(), containsString("unpgradable index"));
```

Comment:
Can we not use hamcrest? I've been trying to eliminate it everywhere I see it...
```
assertTrue(ex.getMessage(), ex.getMessage().contains("unupgradable index");
```

Also, I agree with Shay's previous comment we should have a better exception message. And I'm not even sure unupgradeable is a word...certainly not unpgradable. :)